mypyc/irbuild/specialize.py

```diff
@@ -57,6 +57,25 @@
 specializers: Dict[Tuple[str, Optional[RType]], List[Specializer]] = {}
 
 
+def get_specialization(builder: 'IRBuilder', expr: CallExpr, callee: RefExpr,
+                       typ: Optional[RType] = None) -> Optional[Value]:
+    # TODO: Allow special cases to have default args or named args. Currently they don't since
+    # they check that everything in arg_kinds is ARG_POS.
+
+    if isinstance(callee, (MemberExpr, NameExpr)):
+        name: Optional[str] = callee.fullname if callee.fullname else callee.name
```

Comment:
We should only use `callee.name` if it's a `MemberExpr` and `typ` is defined. I think that this logic is a bit too complicated and it would perhaps be better to have separate functions for `get_function_specialization` and `get_method_specialization` (only the latter would accept `typ` or `receiver_type`).

Reply:
have a look and see if I did this right.